spec/index.md

```diff
@@ -4,11 +4,383 @@
+> &nbsp;&nbsp; `@`[IDENTIFIER]
+>
+> _FunctionParameters_ :\
+> &nbsp;&nbsp; _FunctionParam_ (`,` _FunctionParam_)<sup>\*</sup> `,`<sup>?</sup>
+>
+> _FunctionParam_ :\
+> &nbsp;&nbsp; [IDENTIFIER] `:` [_Type_]
+>
+> _FunctionReturnType_ :\
+> &nbsp;&nbsp; `->` [_Type_]
+
+
+A _function_ consists of a [block], along with a name and a set of parameters.
+Other than a name, all these are optional. Functions are declared with the
+keyword `def`. Functions may declare a set of *input* [*variables*][variables]
+as parameters, through which the caller passes arguments into the function, and
+the *output* [*type*][type] of the value the function will return to its caller
+on completion.
+
+When referred to, a _function_ yields a first-class *value* of the
+corresponding zero-sized [*function item type*], which
+when called evaluates to a direct call to the function.
+
+A function header ends with a colon (`:`) after which the function body begins.
+
+For example, this is a simple function:
+
+```python
+def answer_to_life_the_universe_and_everything() -> u256:
+    return 42;
+```
+
+#### 3.1.1 Visibility and Privacy
+
+These two terms are often used interchangeably, and what they are attempting to convey is the answer to the question "Can this item be used at this location?"
+
+Fe knows two different types of visibility for functions and state variables: `public` and `private`. Visibility of `private` is the default and is used if no other visiblity is specified. 
+
+**Public:** External functions are part of the contract interface, which means they can be called from other contracts and via transactions.
+
+**Private:** Those functions and state variables can only be accessed internally from within the same contract. This is the default visibility.
+
+For example, this is a function that can be called externally from a transaction:
+
+```python
+pub def answer_to_life_the_universe_and_everything() -> u256:
+    return 42;
+```
+
+### 3.2. Structs
+
+> **<sup>Syntax</sup>**\
+> _Struct_ :\
+> &nbsp;&nbsp; `struct`
+>   [IDENTIFIER]&nbsp;
+> &nbsp;&nbsp; [_EndOfHeader_]\
+> &nbsp;&nbsp; _StructField_<sup>\*</sup>
+>
+> _StructField_ :\
+> &nbsp;&nbsp; [IDENTIFIER] `:` [_Type_]
+
+
+A _struct_ is a nominal [struct type] defined with the keyword `struct`.
+
+An example of a `struct` item and its use:
+
+```
+struct Point:
+    x: u256
+    y: u256
+
+p = Point {x: 10, y: 11}
+px: u256 = p.x;
+```
+
+### 3.3. Events
+
+> **<sup>Syntax</sup>**\
+> _Event_ :\
+> &nbsp;&nbsp; `event`
+>   [IDENTIFIER]&nbsp;
+> &nbsp;&nbsp; [_EndOfHeader_]\
+> &nbsp;&nbsp; _EventField_<sup>\*</sup>
+>
+> _EventField_ :\
+> &nbsp;&nbsp; _EventIndexability_ [IDENTIFIER] `:` [_Type_]
+
+> _EventIndexability_ :\
+> &nbsp;&nbsp; `indexed`<sup>?</sup>
+
+An _event_ is a nominal [event type] defined with the keyword `event`. It is emitted with the keyword `emit`.
+
+An example of a `event` item and its use:
+
+```
+event Transfer:
+    indexed sender: address
```

Comment:
@g-r-a-n-t , I guess this should be `idx` instead, right?

Reply:
Yeah, this is what's currently used as I recall.